Lib/test/test_asyncore.py

```diff
@@ -843,5 +845,49 @@ class TestAPI_UseUnixSocketsSelect(TestAPI_UseUnixSockets, unittest.TestCase):
 class TestAPI_UseUnixSocketsPoll(TestAPI_UseUnixSockets, unittest.TestCase):
     use_poll = True
 
+
+class PollTests(unittest.TestCase):
+    def check_poll_modify_map(self, use_poll):
+        testmap = {}
+        read_calls = []
+
+        class Handler:
+            def __init__(self, fd):
+                self.fd = fd
+            def readable(self):
+                return True
+            def writable(self):
+                return False
+            def accepting(self):
+                return False
+            def handle_read_event(self):
+                read_calls.append(self.fd)
+                testmap.clear()
```

Comment:
In the meanwhile, I added a new test_poll_close_replace_dispatchers() test which tests exactly the described behaviour.